src/ripple/app/transactors/Transactor.cpp

```diff
@@ -302,4 +284,356 @@ TER Transactor::apply ()
     return doApply ();
 }
 
+TER Transactor::checkSign ()
+{
+#if RIPPLE_ENABLE_MULTI_SIGN
+    // If the mSigningPubKey is empty, then we must be multi-signing.
+    TER const signingTER = mSigningPubKey.getAccountPublic ().empty () ?
+        checkMultiSign () : checkSingleSign ();
+#else
+    TER const signingTER = checkSingleSign ();
+#endif // RIPPLE_ENABLE_MULTI_SIGN
+
+    return signingTER;
+}
+
+TER Transactor::checkSingleSign ()
+{
+    // Consistency: Check signature
+    // Verify the transaction's signing public key is authorized for signing.
+    if (mSigningPubKey.getAccountID () == mTxnAccountID)
+    {
+        // Authorized to continue.
+        mSigMaster = true;
+        if (mTxnAccount->isFlag(lsfDisableMaster))
+            return tefMASTER_DISABLED;
+    }
+    else if (mHasAuthKey &&
+        (mSigningPubKey.getAccountID () ==
+            mTxnAccount->getFieldAccount160 (sfRegularKey)))
+    {
+        // Authorized to continue.
+    }
+    else if (mHasAuthKey)
+    {
+        m_journal.trace <<
+            "applyTransaction: Delay: Not authorized to use account.";
+        return tefBAD_AUTH;
+    }
+    else
+    {
+        m_journal.trace <<
+            "applyTransaction: Invalid: Not authorized to use account.";
+        return temBAD_AUTH_MASTER;
+    }
+
+    return tesSUCCESS;
+}
+
+namespace TransactorDetail
+{
+
+struct GetSignerListResult
+{
+    TER ter = tefFAILURE;
+    std::uint32_t quorum = std::numeric_limits <std::uint32_t>::max ();
+    std::vector<SignerEntries::SignerEntry> signerEntries;
+};
+
+// We need the SignerList for every SigningFor while multi-signing.
+GetSignerListResult
+getSignerList (
+    Account signingForAcctID, TransactionEngine* engine, beast::Journal journal)
+{
+    GetSignerListResult ret;
+
+    uint256 const index = getSignerListIndex (signingForAcctID);
+    SLE::pointer accountSignersList =
+        engine->view ().entryCache (ltSIGNER_LIST, index);
+
+    // If the signer list doesn't exist the account is not multi-signing.
+    if (!accountSignersList)
+    {
+        journal.trace <<
+            "applyTransaction: Invalid: Not a multi-signing account.";
+        ret.ter = temBAD_AUTH_MASTER;
```

Comment:
I've added a `tefNOT_MULTI_SIGNING` code.  I picked a `tef` code based on this comment in TER.h:
```
    // Causes:
    // - Transaction cannot succeed because of ledger state.
```
I'm using that new code here because the ledger doesn't contain a signer list for the account.